diff --git a/option/tun.go b/option/tun.go
index 72b6e456..bfc24463 100644
--- a/option/tun.go
+++ b/option/tun.go
@@ -44,6 +44,8 @@ type TunInboundOptions struct {
 	Platform                      *TunPlatformOptions              `json:"platform,omitempty"`
 	InboundOptions
 
+	DisableDNSHijack bool `json:"disable_dns_hijack,omitempty"`
+
 	// Deprecated: removed
 	GSO bool `json:"gso,omitempty"`
 	// Deprecated: merged to Address
diff --git a/protocol/tun/inbound.go b/protocol/tun/inbound.go
index 6820831a..60c086ae 100644
--- a/protocol/tun/inbound.go
+++ b/protocol/tun/inbound.go
@@ -199,6 +199,8 @@ func NewInbound(ctx context.Context, router adapter.Router, logger log.ContextLo
 			ExcludePackage:                        options.ExcludePackage,
 			InterfaceMonitor:                      networkManager.InterfaceMonitor(),
 			EXP_MultiPendingPackets:               multiPendingPackets,
+
+			EXP_DisableDNSHijack: options.DisableDNSHijack,
 		},
 		udpTimeout:        udpTimeout,
 		stack:             options.Stack,
